### PR TITLE
Read DQ flags with finer time resolution

### PR DESCRIPTION
**Pull request by maxtrevor**
The performance of DQ flags is much better if we don't have to pad them to start/end on integer seconds. Previously PyCBC only checked if the start of each second was flagged, with no support for better time resolution.

### CODE REVIEW

File: bin/all_sky_search/pycbc_calculate_dqflag [hunk 1]
```diff
@@ -26,6 +26,8 @@ parser.add_argument('--flag', type=str, required=True,
                     help="name of the data quality flag")
 parser.add_argument('--ifo', type=str, required=True,
                     help="interferometer to analyze")
+parser.add_argument('--sample-frequency', required=False,
+                    default=128, help="timeseries sampling frequency")
```
Comment (tdent) [inline bin/all_sky_search/pycbc_calculate_dqflag:30]: Why 128 as default? Seems a rather large value compared to 1 as used before. Do you need a default anyway?

Also suggest adding units to the help ie `(Hz)`

**Review by tdent — CHANGES_REQUESTED** [review]
Seems simple, one comment/suggestion

File: bin/all_sky_search/pycbc_calculate_dqflag (hunk 1 above)
Comment (maxtrevor) [reply bin/all_sky_search/pycbc_calculate_dqflag:30]: Many of the vetoes I want to use have durations as little as 1/4 of a second. The idq timeseries itself has a sample rate of 1024 so theoretically vetoes can start/end on any 1/1024th of a second. 128 felt like a reasonable setting to pick up on a flag starting quickly without actually checking every 1024th of a second. But I can easily make this configurable.

**Review by tdent — APPROVED** [review]
Approved, assuming that it works

**Comment by maxtrevor** [comment]
Removing the default option exposed a bug when the sample frequency was interpreted as a string by default. Had to add a type specification to the command line option

**Pull request merged**